### ISSUE TITLE
"leave conversation" request has empty token

### ISSUE THREAD

**Issue by mahibi** [nextcloud/talk-android#1633]
reported via chat. no more details available.

### Expected behaviour
DELETE /ocs/v2.php/apps/spreed/api/v4/room/HERE_THE_TOKEN/participants/active HTTP/1.1

### Actual behaviour
DELETE /ocs/v2.php/apps/spreed/api/v4/room//participants/active HTTP/1.1

**Comment by mahibi** [comment]
to debug from here:

https://github.com/nextcloud/talk-android/blob/e0ecf049b39a1dc9dfe618322033cf79830df741/app/src/main/java/com/nextcloud/talk/utils/ApiUtils.java#L197

my first guess is that the empty token might be caused by the default value here:
https://github.com/nextcloud/talk-android/blob/master/app/src/main/java/com/nextcloud/talk/controllers/ChatController.kt#L269

**Comment by mahibi** [comment]
@nickvergessen can you think of any bugs that this empty token might cause?

**Comment by nickvergessen** [comment]
Yes, the session is never removed on the server, so this could be one of the reasons when people "don't get notifications".
Situation was fixed a bit on the server by ignoring sessions which are older than 30 seconds, but removing the session with the leave would reduce the risk even more.

**Comment by mahibi** [comment]
see
https://github.com/nextcloud/talk-android/blob/master/app/src/main/java/com/nextcloud/talk/activities/CallActivity.java#L1616

**Comment by AndyScherzinger** [comment]
No activity since 2021 and room join and leave handling were rewritten with the chat rewrite. Closing as stale — please open a fresh report if the leave request is still sent with an empty token on 25.x.
